### PR TITLE
Plinq ActiveIssue tests

### PR DESCRIPTION
**Pull request by Clockwork-Muse**
This PR is to add tests covering expected behavior that is part of current issues (as well as some minor missing ones).  
Some of these tests are now marked `ActiveIssue`, and will not succeed against the current implementation.  They have been tested to succeed if the current implementation is corrected as I currently expect.

The issues covered by these tests (but not fixed!) are:
#1330
#1332

@stephentoub , @AlfredoMS

### CODE REVIEW

**Comment by AlfredoMS** [comment]
@Clockwork-Muse thanks for the good work as always, I'll be reviewing this pr and the related issues very soon.

**Comment by AlfredoMS** [comment]
#1330 was closed, but #1332 is still under review.

sorry to have make you wait so long.

test LGTM, @stephentoub any objections?

File: src/System.Linq.Parallel/tests/Combinatorial/ParallelQueryCombinationTests.cs [hunk 1]
```diff
@@ -133,6 +133,36 @@ public static void Cast_NotPipelined(LabeledOperation source, LabeledOperation o
         }
 
         [Theory]
+        [ActiveIssue(1332)]
```
Comment (stephentoub) [inline src/System.Linq.Parallel/tests/Combinatorial/ParallelQueryCombinationTests.cs]: Let's remove the tests marked as [ActiveIssue(1332)] or [ActiveIssue(1330)]

File: src/System.Linq.Parallel/tests/Combinatorial/ReverseOrderedParallelQueryCombinationTests.cs [hunk 2]
```diff
@@ -0,0 +1,851 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Xunit;
+
+namespace System.Linq.Parallel.Tests
+{
+    public partial class ParallelQueryCombinationTests
+    {
+        [Theory]
+        [Trait("category", "reverse")]
```
Comment (stephentoub) [inline src/System.Linq.Parallel/tests/Combinatorial/ReverseOrderedParallelQueryCombinationTests.cs]: Please remove these custom traits

File: src/System.Linq.Parallel/tests/Combinatorial/ReverseOrderedParallelQueryCombinationTests.cs [hunk 3]
```diff
@@ -0,0 +1,851 @@
...
+        [MemberData("UnaryReversedOperators")]
+        [MemberData("BinaryReversedOperators")]
```
Comment (stephentoub) [inline src/System.Linq.Parallel/tests/Combinatorial/ReverseOrderedParallelQueryCombinationTests.cs]: Please use nameof for all of these member datas

**Comment by stephentoub** [comment]
Some additional comments and things to be cleaned up since this was opened, but otherwise the new tests LGTM.

**Comment by Clockwork-Muse** [comment]
@stephentoub - the reverse tests were originally mostly to cover for what was happening in the second part of 1332 (single ordered source, not same as generated order), but that was marked as expected.  Should I remove them altogether, or it's something we should keep?  I do have tests dealing with reversed data already, to an extent at least.

**Comment by stephentoub** [comment]
> Should I remove them altogether, or it's something we should keep? I do have tests dealing with reversed data already, to an extent at least.

If they're not adding value, let's remove them.

**Comment by Clockwork-Muse** [comment]
@stephentoub , @AlfredoMS  - updated.  I ended up rebasing to current because it was getting a little stale.

Given that #1155 was labeled as "bug", I went ahead and added the relevant regression tests to this (last two commits).  If necessary I can pull them.

**Comment by Clockwork-Muse** [comment]
Strange, CentOS failed, but nothing's listed in the logs...

@dotnet-bot - Test Innerloop CentOS7.1 Release please

**Comment by Clockwork-Muse** [comment]
@dotnet-bot Innerloop CentOS7.1 Release Build and Test please

**Comment by Clockwork-Muse** [comment]
@dotnet-bot test Innerloop CentOS7.1 Release Build and Test

**Comment by stephentoub** [comment]
Test Innerloop OSX Debug Build and Test Please
Test Innerloop OSX Release Build and Test Please

**Comment by stephentoub** [comment]
LGTM.  Thanks for the additional tests, @Clockwork-Muse.

**Pull request merged**